test/tests/functional/pbs_hook_execjob_end.py

```diff
@@ -257,3 +257,74 @@ def test_execjob_end_delete_request(self):
                           interval=2)
             msg = "Got expected log_msg on host:%s" % host
             self.logger.info(msg)
+
+    def test_execjob_end_reject_request(self):
+        """
+        Test to make sure hook job reject message should appear in mom log
+        in case sister mom went down before executing execjob_end hook
+        """
+
+        if len(self.moms) != 2:
+            self.skip_test(reason="need 2 mom hosts: -p moms=<m1>:<m2>")
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+
+        # Create hook
+        hook_name = "execjob_end_logmsg7"
+        self.server.create_import_hook(hook_name, self.attr, self.hook_body)
+
+        # Submit a multi-node job
+        a = {'Resource_List.select':
+             '1:ncpus=1:host=%s+1:ncpus=1:host=%s' %
+             (self.momA.shortname, self.momB.shortname),
+             'Resource_List.place': 'excl'}
+        j = Job(TEST_USER, attrs=a)
+        j.set_sleep_time(60)
+        jid = self.server.submit(j)
+
+        # Verify job spawn on sisterm mom
+        self.momB.log_match("Job;%s;JOIN_JOB as node" % jid, n=100,
+                            max_attempts=10, interval=2)
+
+        # When the job run approx 10-20 sec, bring sister mom down
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid, offset=20)
+        msg = 'mom is not down'
+        self.assertTrue(self.momB.stop(), msg)
+
+        # Verify momB is down and job is running
+        self.server.expect(
+            NODE, {'state': "down,job-exclusive"}, id=self.momB.shortname)
```

Comment:
I am not sure why do you need exclusive job. you are only submitting one job. but that is okay. how about you just match that node state is down and ignore any other node state? I believe for your test it doesn't matter if node is job-exclusive or not unless I am missing something.

Reply:
I don't really need exclusive job my purpose is to just confirm node should be in down  state, updated test as you suggested.